src/rgw/rgw_asio_frontend.cc

```diff
@@ -451,50 +451,43 @@ int AsioFrontend::init_ssl()
   // ssl configuration
   auto cert = config.find("ssl_certificate");
   const bool have_cert = cert != config.end();
-  if (have_cert) {
-    // only initialize the ssl context if it's going to be used
-    ssl_context = boost::in_place(ssl::context::tls);
+  if (!have_cert) {
+    lderr(ctx()) << "no ssl_certificate configured for ssl_port" << dendl;
+    return -EINVAL;
```

Comment:
AsioFrontend::init() calls init_ssl() unconditionally - returning this error would prevent us from starting any non-ssl endpoints too. the goal of this logic is to only require an ssl_certificate if an ssl port/endpoint is actually requested

Reply:
makes sense, I just added a check before we process endpoints/ports in the v6 pr https://github.com/ceph/ceph/pull/27270